src/mesh/index_derivs.cxx

```diff
@@ -570,91 +548,79 @@ Mesh::flux_func sfFDDX, sfFDDY, sfFDDZ;
  *******************************************************************************/
 
 /// Set the derivative method, given a table and option name
-void derivs_set(Options *options, DiffLookup *table, const char *name,
-                Mesh::deriv_func &f) {
-  TRACE("derivs_set( deriv_func )");
-  string label;
-  options->get(name, label, "C2");
-
-  DIFF_METHOD method = lookupFunc(table, label); // Find the function
-  printFuncName(method);                         // Print differential function name
-  f = lookupFunc(table, method);                 // Find the function pointer
-}
-
-void derivs_set(Options *options, DiffLookup *table, const char *name,
-                Mesh::upwind_func &f) {
-  TRACE("derivs_set( upwind_func )");
-  string label;
-  options->get(name, label, "U1");
-
-  DIFF_METHOD method = lookupFunc(table, label); // Find the function
-  printFuncName(method);                         // Print differential function name
-  f = lookupUpwindFunc(table, method);
-}
+template <typename T, typename Ts>
+void derivs_set(std::vector<Options *> options, DiffLookup *table, DiffLookup *stable,
+                const std::string &name, const std::string &def, T &f, Ts &sf,
+                bool staggerGrids) {
+  TRACE("derivs_set()");
+  output_info.write("\t%-12s: ", name.c_str());
+  string label = def;
+  for (auto &opts : options) {
+    if (opts->isSet(name)) {
+      opts->get(name, label, "");
+      break;
+    }
+  }
 
-void derivs_set(Options *options, DiffLookup *table, const char *name,
-                Mesh::flux_func &f) {
-  TRACE("derivs_set( flux_func )");
-  string label;
-  options->get(name, label, "U1");
+  f = lookupFunc(table, label); // Find the function
 
-  DIFF_METHOD method = lookupFunc(table, label); // Find the function
-  printFuncName(method);                         // Print differential function name
-  f = lookupFluxFunc(table, method);
+  label = def;
+  if (staggerGrids) {
+    output_info.write("\tStag. %-6s: ", name.c_str());
+    for (auto &_name : {name + "stag", name}) {
+      for (auto &opts : options) {
+        if (opts->isSet(_name)) {
+          opts->get(_name, label, "");
+          goto endloop;
```

Comment:
Is the `goto` necessary here? Could you not just do

```cpp
    opts->get(_name, label, "");
    sf = lookupFunc(stable, label);
    return;
```
Am I missing something?

Reply:
As it only duplicates one line of code, it should be fine ...
In general `goto`s are underrated - they can be quite useful - e.g. for breaking loops :)